### PR TITLE
Add functionality to edit/delete post on separate page

### PR DESCRIPTION
**Pull request by GribinetDaniel**

### CODE REVIEW

File: packages/user-interface/src/components/DeletePostModal/DeletePostModal.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <Modal title='Are you sure?'>
+      <ModalContent>
+        <div className='modal__confirm-text'>
+          Do you really want to delete this post?
+          <br />
+          This procces can not be undone!
+        </div>
+      </ModalContent>
+      <ModalFooter>
+        <button
```
Comment (H7GhosT) [inline packages/user-interface/src/components/DeletePostModal/DeletePostModal.tsx]: Create a component for button, and control whether it's type is 'primary' or 'secondary' through it's props

**Review by H7GhosT — CHANGES_REQUESTED** [review]

File: packages/user-interface/src/pages/EditPost/EditPost.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import { PostContent } from '../../components/PostContent';
+import { defaultPost, Post } from '../../types';
+import { isAxiosError } from 'axios';
+
+export function EditPost() {
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+  const path = window.location.pathname;
+  const { isLoading, error, data } = useQuery(
+    `edit-post`,
```
Comment (H7GhosT) [inline packages/user-interface/src/pages/EditPost/EditPost.tsx:24]: Try to edit one post, then right after that go to edit another post and see what happens
Do not refresh the page while doint this

File: packages/user-interface/src/pages/EditPost/EditPost.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setPost({ ...post, [event.target.name]: event.target.value });
+  };
+
+  const handleSubmit = async (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    try {
+      await mainAxios.patch(`${path}`, post);
```
Comment (H7GhosT) [inline packages/user-interface/src/pages/EditPost/EditPost.tsx]: Use mutations here and set loading to action buttons

File: packages/user-interface/src/pages/Register/Register.tsx [hunk 4]
```diff
@@ -64,14 +58,16 @@ export function Register() {
...
-        let errs: Array<any> = [];
-        const obj: Err = {};
-        errs = err.response.data.errors;
-        errs.forEach((err) => (obj[err.param] = err.msg));
-        setErrors(obj);
-        if (obj.username || obj.name || obj.email || obj.password)
-          setCurrentStep(0);
+      } catch (err) {
+        if (isAxiosError(err)) {
+          let errs: Array<{ msg: string; param: string }> = [];
```
Comment (H7GhosT) [inline packages/user-interface/src/pages/Register/Register.tsx]: Create an util for extracting error messages

File: packages/user-interface/src/components/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './index.scss';
+interface ButtonProps {
+  text: string;
```
Comment (H7GhosT) [inline packages/user-interface/src/components/Button/Button.tsx:4]: You can pass a `children` prop and that will get the elements passed here <Button>{children}</Button>

Comment (H7GhosT) [reply packages/user-interface/src/components/Button/Button.tsx:4]: Guess what type you need to set for children

File: packages/user-interface/src/components/DeletePostModal/DeletePostModal.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Modal, ModalContent, ModalFooter } from '../Modal';
+import { Button } from '../Button';
+
+interface DeletePostModalProps {
+  deletePost?: (e: React.SyntheticEvent) => void;
+  setShowModal: (arg: boolean) => void;
```
Comment (H7GhosT) [inline packages/user-interface/src/components/DeletePostModal/DeletePostModal.tsx]: For event handlers use namings like `onDelete`, `onClose`. And you don't need paramenters here

File: packages/user-interface/src/components/Modal/Modal.tsx [hunk 7]
```diff
@@ -4,13 +4,12 @@ import './index.scss';
 export interface ModalProps {
   title: string;
   children: React.ReactNode;
-  style?: any;
 }
 
-export function Modal({ children, title, style }: ModalProps) {
+export function Modal({ children, title }: ModalProps) {
   return (
     <div className='modal'>
```
Comment (H7GhosT) [inline packages/user-interface/src/components/Modal/Modal.tsx:12]: Does your modal close when you click outside it?

File: packages/user-interface/src/components/UserCard/UserCard.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { User } from '../../types';
+import './index.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrashCan } from '@fortawesome/free-regular-svg-icons';
+import { faUserPen } from '@fortawesome/free-solid-svg-icons';
+
+interface UserCardProps {
+  user: User;
+  setShowModalEdit: (arg: boolean) => void;
```
Comment (H7GhosT) [inline packages/user-interface/src/components/UserCard/UserCard.tsx]: Same here `onEdit`, `onDelete`....

**Review by H7GhosT — CHANGES_REQUESTED** [review]

File: packages/user-interface/src/components/Button/Button.tsx (hunk 5 above)
Comment (Obi1kan) [reply packages/user-interface/src/components/Button/Button.tsx:4]: > Guess what type you need to set for children

ReactNode?

File: packages/user-interface/src/pages/EditPost/EditPost.tsx (hunk 2 above)
Comment (GribinetDaniel) [reply packages/user-interface/src/pages/EditPost/EditPost.tsx:24]: All it's good. Do you mean that in useQuery the key is always 'edit-post'

**Pull request merged**